core/src/replay_stage.rs

```diff
@@ -3046,6 +3047,51 @@ impl ReplayStage {
                     }
                 }
 
+                if bank.collector_id() != my_pubkey {
+                    // If the block does not have at least DATA_SHREDS_PER_FEC_BLOCK shreds in the last FEC set,
+                    // mark it dead. No reason to perform this check on our leader block.
+                    if !blockstore
+                        .is_last_fec_set_full(bank.slot())
+                        .inspect_err(|e| {
+                            warn!(
+                                "Unable to determine if last fec set is full for slot {} {},
+                                marking as dead: {e:?}",
+                                bank.slot(),
+                                bank.hash()
+                            )
+                        })
+                        .unwrap_or(false)
```

Comment:
Since now this is marked as `dead` instead of duplicate, I am wondering if we should treat `Err(...)` different from `Ok(false)`.

Reply:
I think if the database operations fail then we should definitely not vote on this block. 
We'll have plenty of soak time before the feature flag is activated to rule out implementation errors in `is_last_fec_set_full`